docs/guide/exe_verilator.rst

```diff
@@ -958,6 +958,22 @@ Summary:
    delayed assignments.  This option should only be used when suggested by
    the developers.
 
+.. option:: --output-groups <files>
+
+   Enables concatenating the output .cpp files into the given number of
+   effective output groups.  This is useful if the compiler startup overhead
+   cumulated from compiling many small files becomes unacceptable, which can
+   happen in designs making extensive use of SystemVerilog classes, templates
+   or generate blocks.  Incompatible with :vlopt:`--output-split`.
```

Comment:
We should allow use with output-split (as discussed), and please add a test with this.  E.g. we split so nothing is too big, then combine so not too many files.  So really small designs might get fewer files, but larger designs should end up with ~~ numfiles of equal-ish size.

Reply:
Added a test copied from `t_flag_csplit`, but with `--output-groups` specific checks added. Also, switched away from `--no-skip-identical` by just checking the Makefile (faster test re-runs).